Checkpoint 4 Review

lewisemm opened the pull request:
Pull request for checkpoint 4 review

lewisemm committed:
(chore) Create project plan

lewisemm committed:
(feature) Create vistagrid project

lewisemm committed:
(feature) Create vistagrid app

lewisemm committed:
(chore) Install bower

lewisemm committed:
(chore) Create app resources directory

lewisemm committed:
(chore) Ignore project dependencies and compiled files

lewisemm committed:
(feature) Create index page

- Configured template directories
- Created a view that returns a response to the index page
- Configured urls to route to the index view
- Created the index view

lewisemm committed:
(chore) Update configuration files

- Update gitignore file
- Create requirements file

lewisemm committed:
(feature) Create login page

lewisemm committed:
(chore) Ignore sqlite and .env files

lewisemm committed:
(feature) Intergrate python-social-auth

lewisemm committed:
(chore) Update readme

lewisemm committed:
(feature) Create Logout button

lewisemm committed:
(feature) Create LoginController

lewisemm committed:
(feature) Create dashboard page template

lewisemm committed:
(feature) Create dashboard mockup

lewisemm committed:
Merge pull request #1 from andela-lkabui/feature/dashboard

Feature/dashboard

lewisemm committed:
(feature) Create Photo model

lewisemm committed:
Merge pull request #3 from andela-lkabui/feature/models

(feature) Create Photo model

lewisemm committed:
(feature) Create RESTful service

- Installed and configured djangorestframework, created serializer and
viewset
- Modify Photo model by adding photo id and changing FileField to
ImageField
- Create new migrations and set MEDIA_URL and MEDIA_ROOT to '/uploads/'
url and 'uploads/' file path respectively

lewisemm committed:
Merge pull request #4 from andela-lkabui/feature/rest_api

(feature) Create RESTful service

lewisemm committed:
(feature) Include Photo model in admin site

lewisemm committed:
(feature) Display uploaded images on dashboard

- Configure vistagrid.urls to display image url is requested
- Create service to query API to retrieve images uploaded images by the
current user
- Use PhotoService on DashboardController to retrieve uploaded files
- Alter HTML to display uploaded images
- Include the new service and controller as a script on the base.html
file

lewisemm committed:
Merge pull request #5 from andela-lkabui/feature/display_uploaded_images

Feature/display uploaded images

lewisemm committed:
(feature) Create logic to apply filters on images

- Create choice fields in the PhotoSerializer to display
available effects as choices on the viewset
- Create a dictionary in Photo model to associate string
representations of effects from the serializer with actual filter
classes from Pillow
- Create custom retrieve and update methods for the PhotoViewSet
to display thumbnails of effects on the dashboard and to apply effects
on photos when they are specified

lewisemm committed:
(feature) Create logic to respond to clicks on a photo

- Create logic for dashboard to display the clicked photo on the main
display area

lewisemm committed:
(chore) Ignore uploads directory

lewisemm committed:
(fix) Delete photo files on disk when a delete request is sent

lewisemm committed:
(feature) Create Effects model

- Create Effects model to enable the creation of a REST API route to
retrieve/create effect previews of a specified upload
- Use Django signals to remove file photos on disk whenever the
'delete()' method is called on an Effect instance/queryset
- Use the 'effect_file_name' method to return the upload path when it's
called by the 'upload_to' kwarg of path attribute in Effects model
- Abstract 'FILTERS' dictionary to reside outside the models so that
it can easily be referred to by both Photo and Effects models

lewisemm committed:
(feature) Create serializer for Effects model

lewisemm committed:
(chore) Replace spaces with tabs

lewisemm committed:
(feature) Create EffectsViewSet

- Use faker to assign random file names to be used when photo preview
files are being created on disk. Random file names are preferred so as
to trigger a refresh on the client side whenever new thumbnails are
created.
- Create logic to handle cases where the 'path' argument is a string
(a file's url) instead of an instance of InMemoryUploadedFile. When a
file url string is used, a file object is created from that and then
passed on to the serializer through the data dictionary
- When POST is called on this route, the number of thumbnails created
will correspond to the number of effects/filters this app currently
has to offer

lewisemm committed:
(chore) Create migrations

lewisemm committed:
(chore) Enforce standard size for previews for consist display in view

lewisemm committed:
(chore) Update bower and virtual env dependencies

lewisemm committed:
(feature) Style app view. Change Theme colours

lewisemm committed:
(feature) Use CSRF tokens. Inject ng-file-upload dependency

lewisemm committed:
(feature) Create view responsiveness logic

- Abstract code to generate thumbnails to a function to allow for code
reuse when uploaded photos are clicked.
- Abstract code to fetch uploaded photos to a function to allow for
code reuse when an image is uploaded
- Create logic to upload new photos from the view

lewisemm committed:
(feature) Add more functionality to PhotoService

- Create POST functionality on Uploads resource
- Add Thumbnails resource to create/retrieve effect preview thumbnails

lewisemm committed:
(feature) Modify dashboard aesthetics

- Change theme colours
- Retrieve effect previews from the backend
- Use ng-upload-file package to facilitate file uploads

lewisemm committed:
(chore) Include app dependencies

- Include ng-upload-file scripts
- Include reference to Materialize icons

lewisemm committed:
(chore) Replace spaces with tabs

lewisemm committed:
(feature) Display effects on main display area when preview is clicked

lewisemm committed:
(chore) Install and include sweetalert in app

lewisemm committed:
(feature) Create behaviour for Save and Delete button

lewisemm committed:
(fix) Specify redirect url after successful login on Facebook

lewisemm committed:
(feature) Define logic to handle deletion of uploaded photos

- Create method deletePhoto to handle delete requests
- Wire up delete functionality on dashboard view
- Change delete function element id

lewisemm committed:
(feature) Create delete method on Uploads resource of PhotoService

lewisemm committed:
Merge pull request #6 from andela-lkabui/feature/responsive_dashboard

Feature/responsive dashboard

lewisemm committed:
(feature) Display file name as part of REST response

- Create get_file_name method in Photo model to return filename from
the model's path attribute
- Create file_name field in PhotoSerializer to refer to get_file_name
method in Photo model. This file_name field is used by the view when
displaying file names alongside uploaded images

lewisemm committed:
(fix) Remove retrieve method from PhotoViewSet

- Remove retrieve field so as to display the path as a http url for use
when sharing a photo through a public url on Facebook

lewisemm committed:
(feature) Create a share button for public Facebook URLs

- Include Facebook SDK to allow the use of share dialogs while sharing
photo urls
- Style uploaded images column with materialize image cards and custom
borders and margins
- Change class name from maroon_icon to facebook_blue_icon to enhance
inference

lewisemm committed:
(feature) Create function to share photos via url on Facebook

lewisemm committed:
(feature) Refine UI controls

- Replace save and delete buttons with materialize fixed action buttons
- Replace img element with materialize card to display selected photos.
- Style dashboard controls with font awesome icons
- Use max-height and height properties at materialize card main display
to guarantee consistent display of photos despite image size

lewisemm committed:
(fix) Add response messages to provide action feedback

lewisemm committed:
(fix) Remove image resize when creating effect previews

- Removed since view is more tolerant of varying image sizes

lewisemm committed:
(fix) Use fixed height to display uploaded photos

- Use max-height and max-width to ensure consistent display of
photos despite photo size

lewisemm committed:
(chore) Stylize Upload button

lewisemm committed:
Merge pull request #7 from andela-lkabui/style-ui

Style ui

lewisemm committed:
(fix) Redirect to root when invalid route is requested

lewisemm committed:
(feature) Protect dashboard view

- Re-route to '/login' when unauthenticated request to '/dashboard'
is made
- Create FacebookService to query the Facebook API to determine if
the user is connected (or not)

lewisemm committed:
(chore) Update app dependencies

- Remove LoginController (controller deleted)
- Include FacebookService

lewisemm committed:
(chore) Style dashboard elements

lewisemm committed:
Merge pull request #8 from andela-lkabui/view-access

View access

lewisemm committed:
(feature) Create initial tests

lewisemm committed:
(feature) Create permissions

- Create IsOwner permissions that restrict access to owner (uploader)
of the photo
- Adjust PhotoViewSet by overriding get_object method that ensures that
IsOwner permissions are applied.
- Override get_queryset method of PhotoViewSet to ensure that the appr-
opriate querysets are returned for both 'list' and 'retrieve' views

lewisemm committed:
(feature) Create tests for Effects route

lewisemm committed:
(chore) Update config files

- Ignore coverage reports
- Ignore virtual environment and migration files when running tests

lewisemm committed:
(feature) Update README

lewisemm committed:
Merge pull request #9 from andela-lkabui/feature/read-me

(feature) Update README

lewisemm committed:
Merge pull request #10 from andela-lkabui/tests

Tests

lewisemm committed:
(feature) Configure Travis CI

lewisemm committed:
(feature) Include coveralls and quantified code badges

lewisemm committed:
(chore) Update requirements file

lewisemm committed:
(fix) Resolve python 3+ errors

lewisemm committed:
(chore) Update requirements file

lewisemm committed:
(fix) Remove Python 2.7 from travis config file

- Remove testing on python2.7 because of urllib.parse ImportError. The
link below leads to a PR that has already been made by a github user to
 address this.

https://github.com/necaris/python-openid/commit/53ec8b0eeb28ed19220df2577b429168698d96e4

lewisemm committed:
Merge pull request #11 from andela-lkabui/feature/continous-intergration

Feature/continous intergration

lewisemm committed:
(chore) Add Travis CI badge to readme

lewisemm committed:
(feature) Configure app to deploy on Heroku

lewisemm committed:
(fix) Restrict photo access to photo owners

lewisemm committed:
(feature) Use PostgreSQL on Heroku

lewisemm committed:
(feature) Deny access to dashboard when user is not logged in

- Create a view to check the authenticated status from the request
object to determine if a user is logged in or not
- If user is not logged in and attempts to access the dashboard, they
will be redirected to login with a message informing them that they
need to authenticate via Facebook
- Create route to access is_logged_in method view
- Create an AuthService that will query the loginstatus and provide
the response that will be used to determine whether the app should
redirect to login

lewisemm committed:
Merge pull request #12 from andela-lkabui/ft-deny-unauthorized-user

(feature) Deny access to dashboard when user is not logged in

lewisemm committed:
(chore) Update dependencies

lewisemm committed:
(feature) Create LoginController for animated text at landing page

lewisemm committed:
(chore) Use Frederika google font

lewisemm committed:
(feature) Display instruction messages on the dashboard

- Display relevant instruction messages depending on whether the
user has previously uploaded photos or not

lewisemm committed:
(fix) Include Facebook JavaScript SDK

lewisemm committed:
Merge pull request #13 from andela-lkabui/ch-change-dashboard-placeholder

Ch change dashboard placeholder

lewisemm committed:
(fix) Fix Logout button

- Use ngCookies to store cookies upon successful login. This cookie
will be used to influence when the logout button (and the logo at the
top right corner) will be displayed
- Both of these are going to be displayed only when a user is logged
in the system

lewisemm committed:
(chore) Add landing page background image

lewisemm committed:
Merge pull request #14 from andela-lkabui/bg-fix-logout-button

Bg fix logout button

lewisemm committed:
Merge pull request #14 from andela-lkabui/bg-fix-logout-button

Bg fix logout button

andela-anandaa commented on README.md:
```diff
@@ -1,2 +1,105 @@
+## Installation and setup
+*  Navigate to a directory of choice on `terminal`.
+*  clone this repository on that directory.
+  *  Using ssh;
```
Use capital SSH and HTTP to refer to the protocols.

andela-anandaa commented on README.md:
```diff
@@ -1,2 +1,105 @@
+
+## Installation and setup
+*  Navigate to a directory of choice on `terminal`.
+*  clone this repository on that directory.
```
"Clone"

andela-anandaa commented on README.md:
````diff
@@ -1,2 +1,105 @@
+*  Install the app's front end dependencies using bower.
+  *  `./node_modules/bower/bin/bower install`
+
+  ``` In order to use bower, you need to install it through **npm**. You also need to have **node** and **git** installed on your system. ```
````
For a _note_ use pull-quote:

> In order to use bower, you need to install it through **npm**. You also need to have **node** and **git** installed on your system.

andela-anandaa commented on app/models.py:
```diff
@@ -0,0 +1,84 @@
+from __future__ import unicode_literals
+
+import os
```
**PEP8:** This import comes first before the `from [this] import [that]` ones.

lewisemm replied:
I tried this but Python insists (through errors) that `from __future__ import unicode_literals` should always be the first line in a file

andela-anandaa commented on app/models.py:
```diff
@@ -0,0 +1,84 @@
+	def save(self, *args, **kwargs):
+		"""Apply the effects to the file on disk whenever model.save() is called."""
+		super(Effects, self).save(*args, **kwargs)
+		# This method is called after save because the 'path' attribute will refer
```
**Nitpick:** You can combine all these in the _docstring_ above.

andela-anandaa commented on app/urls.py:
```diff
@@ -0,0 +1,15 @@
+
+urlpatterns = [
+	url(r'^', include(router.urls)),
+    url(r'^loginstatus', views.is_logged_in, name='loginstatus')
```
**Nitpick:** `loginstatus` is not a good URL naming convention, you could go with `login/status`.

andela-anandaa commented on app/views.py:
```diff
@@ -0,0 +1,28 @@
+
+@api_view(['GET'])
+def is_logged_in(request):
+    """Method view to check if request is coming from a user who is logged in
```
**PEP8:* the closing `"""` for the _docstring_ should come on it's own line.

andela-anandaa commented on app/views.py:
```diff
@@ -0,0 +1,28 @@
+@api_view(['GET'])
+def is_logged_in(request):
+    """Method view to check if request is coming from a user who is logged in
+    through a Django backends authentication system."""
```
**PEP8:** the closing `"""` for the _docstring_ should come on it's own line.

andela-anandaa commented on app/viewsets.py:
```diff
@@ -0,0 +1,145 @@
+
+
+class EffectViewSet(viewsets.ModelViewSet):
+	"""Handle CRUD requests to '/effects/' url."""
```
**PEP8:** Same to this, the closing `"""` should be on it's own line. Correct this and the rest.

lewisemm committed:
(feature) Re-design login page

- Use the less is more principle. Utilize wallpaper with negative space
and style login button and app's description according to PR review
recommendations

lewisemm committed:
(feature) Style main display area to handle different image sizes

- Remove images and display images as backgrounds with contain
(background-size), no-repeat and center (background-position)
properties to display images of different dimensions without affecting
the dashboard layout

lewisemm committed:
(feature) Use Materialize Fixed Action Buttons

- Use materialize fixed action buttons to toggle the share, delete and
save buttons
- Change the app ID to test sharing on the test app
- Apply CSS on added fixed action buttons

lewisemm committed:
(feature) Display effects as thumbnail previews

- Create view of thumbnails that is horizontally scrollable.
- Also display the effect name beneath the effect preview

lewisemm committed:
(feature) Display username and profile picture

- Create SocialAuthUsersocialauth model to query table
social_auth_usersocialauth (which is created by python-social-auth)
so as to provide a means of retrieving the facebook uid
- Refactor app.views to display additional username and uid information
for logged in users
- Update url mapping of app.views according to PR nitpick recommendation
- Update AuthService's url reference to loginStatus url
- Fetch URL of the currently logged in user if the call to AuthService
resolves successfully. This URl is being stored in 'profile_picture'
scope variable. The username from this call is also being stored in
'username' scope variable
- The dashboard has two elements that refer to the 'profile_picture'
and 'username' scope variable. These values are displayed on the dash-
board view
- Create a materialize preloader for uploading photos on the dashboard

lewisemm committed:
(feature) Create Logout functionality

- Create a logout button alongside user's name and profile picture
- Initialize the logout_url to a scope variable at the angular_base html
file
- Create logic to redirect to dashboard when landing page is visited
while a user is logged in
- Create a logout function at the dashboard controller to remove
'isLoggedIn' cookie and to set showLogoutButton to false

lewisemm committed:
(feature) Style upload button

- Design the upload button as the call to action button

lewisemm committed:
(feature) Use materialize preloaders

- Use materialize preloaders during photo uploads and when generating
effect previews to provide feedback to the user that a process is in
progress

lewisemm committed:
(feature) Refactor PhotoViewset to return most recent uploads first

- Use django order_by method to change queryset order (by photo_id desc)
so that most recent photos are displayed on top. Also helps when new
photos are uploaded because the user can see them once the upload
operation is complete

lewisemm committed:
(feature) Refine dashboard view to include app theme colours

- Create placeholder logo and text instructions when no photo has been
clicked yet.
- Replace custom header with materialize nav bar that features the drop
down button that is desired for the logout button
- Change the favicon to reflect change in logo
- Apply css effects on nav bar and materialize preloaders to maintain a
consistent theme of colours
- Use scope variable flag to switch between photo edit view and
placeholder view (i.e. when no photo has yet been clicked) in the
controller

lewisemm committed:
(feature) Use materialize tabs

- Use materialize tabs to switch between 'upload photo' and 'recent
edits' views

lewisemm committed:
Merge branch 'develop' of github.com:andela-lkabui/checkpoint4 into develop

lewisemm committed:
(feature) Restyle login page

- Overhaul login page and eliminate superfluous files in image folder
and css selectors in login.css

lewisemm committed:
Merge pull request #16 from andela-lkabui/restyle-ui

Restyle ui

lewisemm committed:
Merge branch 'develop' of github.com:andela-lkabui/checkpoint4 into develop

lewisemm committed:
(feature) Create a 404 page

- Create a 404 page that displays when an invalid django route is
requested. 404 pages require that the DEBUG variable of the settings
file be set to False
- When DEBUG is false, the server fetches static files (css, images,
html etc) from STATIC_ROOT. (collectstatic has to be run on the server
first)
- Create the 404 page's css and load it in the base.html file

lewisemm committed:
Merge pull request #17 from andela-lkabui/ft-404-page

Ft 404 page

lewisemm committed:
(chore) Update README

- Update README to include additional app dependencies
- Address nitpicks identified during PR review

lewisemm committed:
(chore) Fix QuantifiedCode docstring issues

- Create and format docstrings in PEP257 standards

lewisemm committed:
(chore) Fix QuantifiedCode code quality issues

- Create __srt__ methods for models
- Use pythonic variable name in app/views.py
- Document functions

lewisemm committed:
Merge pull request #18 from andela-lkabui/ch-resolve-qc-issues

Ch resolve qc issues

lewisemm committed:
(feature) Create PhotoEdit model

- Create a PhotoEdit model that persists edits made to the Photo model.
This model stores its associated files in the '/uploads/edits/' folder.
- Create a serializer, viewset and route for the PhotoEdit model

lewisemm committed:
(feature) Create method get_file_name in PhotoEdits model

- Create method get_file_name to provide the filename associated with
the edited photo
- Create a filename attribute in the serializer which gets its value
from the get_file_name method in PhotoEdits
- Set the filename as a tooltip on the dashboard view when a recent
edit is being displayed at the main area.

lewisemm committed:
(feature) Show recent edits on dashboard

- Create a resource to fetch recent edits from the backend
- Create functions and flags in the controller to control display of
recent edits on the dashboard view
- Create tooltips for buttons on the dashboard view and for the clicked
recent edit

lewisemm committed:
Merge pull request #19 from andela-lkabui/ft-recent-edits

Ft recent edits

lewisemm committed:
(fix) Redirect to dashboard if user is logged in

- Use the AuthService to query the '/api/login/status/' endpoint to
determine whether a user is logged in or not. If logged in, redirect to
dashboard, else, maintain the current route

lewisemm committed:
Merge pull request #20 from andela-lkabui/fx-redirect-when-logged-in

(fix) Redirect to dashboard if user is logged in

lewisemm committed:
(chore) Order edits in descending order

- Order photo edits starting with the most recent ones first (to
acheive more intuition on the UI)

lewisemm committed:
(feature) Overhaul dashboard view

- Change theme colours of application in dashboard view

lewisemm committed:
Merge pull request #21 from andela-lkabui/ch-change-theme-colours

Ch change theme colours

lewisemm committed:
(chore) Update gitignore

- Ignore app migrations

lewisemm committed:
(chore) Rename models and resolve QuantifiedCode issues

- Rename models to more intuitive names
- Refactor serializers and viewsets to compensate for adjustment in
names
- Refactor urls to create more descriptive api routes
- Refactor tests to compensate for adjustment of model names and routes
- Refactor Angular services and controllers to adjust for changes in
routes and fields in JSON data from the API

lewisemm committed:
(chore) Remove app migrations

- Remove app migrations from version control

lewisemm committed:
Merge pull request #22 from andela-lkabui/ch-clean-up-models

Ch clean up models

lewisemm committed:
(fix) Fix 404 errors

- Configure MEDIA_ROOT and MEDIA_URL in settings file
- Slice path attribute of Photo model to provide valid image file url

lewisemm committed:
(fix) Fix floating horizontal action button

- Apply z-index of 0 to button that was previously floating above header

lewisemm committed:
(fix) Provide responsive switch between recent edits and uploads view

- Use flags to display appropriate 'div' element when one of the two
tabs is selected

lewisemm committed:
(chore) Remove Facebook Service

- Remove FacebookService. All the tasks it was meant to do are being
done by the facebook_sdk file.
- Load app scripts at the bottom of the page

lewisemm committed:
(fix) Clean up files on disk when model instance's deleted is called

- Use signals and receiver functions to delete model instance's
associated photos on disk.

lewisemm committed:
(fix) Create foreign key in Preview model

- Create a foreign key reference to Photo model in the Preview model.
This was done to ensure that each preview has an associated uploaded
photo
- Refactor the serializer to require the photo as a read/write attribute
and change the path attribute to a read only field. This was done to
eliminate redundancy (since with the photo object, a path can be obtained)
- Refactor the create method of the PreviewViewSet to fetch a Photo
object from the database and use this object as a foreign key to all
Preview objects. The Photo object is also used to provide Preview's path
attribute with a value.
- Refactor the DashboardController to send the photo_id as the modified
backend requires
- Fix failing tests as a result of changes to the back end

lewisemm committed:
Merge pull request #23 from andela-lkabui/fx-resolve-bugs-and-errors

Fx resolve bugs and errors

lewisemm committed:
(fix) Seal loophole in logic

- Check to see if an effect has been specified before proceeding to
create a PhotoEdit object(fix) Seal loophole in logic
- Use try except block in destroy method in case the pk provided does
not exist in the database

lewisemm committed:
(feature) Increase test coverage

- Include tests for edge cases to improve percentage of test coverage
- Fix typo in permissions file to enable application of permissions

lewisemm committed:
(fix) Fix Python 3+ errors

- Fix Python 3+ errors that don't appear in Python 2.7

lewisemm committed:
Merge pull request #24 from andela-lkabui/ft-improve-test-coverage

Ft improve test coverage

lewisemm committed:
(fix) Fix file cleanup when model is deleted

- Receiver methods all had the same name and thus had undesired results.
Assign unique name to each method and associated model files can now be
deleted when delete() method is called on an instance
- Remove file clean up logic from viewset
- Take advantage of fix to clean up the disk and remove  the test images
 whenever a test completes its run
- Test the fix by confirming that associated file exists before delete
operation and it doesn't exist after delete has been requested

lewisemm committed:
(fix) Resolve QuantifiedCode issue

- Use pythonic variable name for fb_object

lewisemm committed:
Merge pull request #25 from andela-lkabui/ft-improve-test-coverage

Ft improve test coverage

lewisemm committed:
Update README.md

Add the link to Heroku app

andela-anandaa commented:
`[Before resubmission]`

#### README
- Well written README, just fix the minor nitpicks noted above.
- Good: badges for _QuantifiedCode, Travis and Coveralls_.
- Add a link to the live application on the README.

#### Github Commits
- The commit messages are okay and consistent. However, next time, make sure all the commit messages have at least a body that explains the _what_ and _why_.

#### Comments
- Sufficiently commented.

#### Code Quality
- Okay, only minor issues raised by [QuantifiedCode](https://www.quantifiedcode.com/app/project/54a4decaa92b4d2483d7a1c3c42f79c0).

#### Functionality
- Fix: auto-redirect to dashboard if the user is logged in
- Fix: have a nice 404 page
- Suggestion: save the edited images as new images other than replacing the old.

#### UX/UI
- When uploading, should indicate the user.
- Check out [Nielson's 10 Usability Heuristics for User Interface Design](https://www.nngroup.com/articles/ten-usability-heuristics/)
- Fix the borders as discussed.

#### Testing
- Good: 90% code coverage.
- Good: All the tests are passing.

lewisemm committed:
(fix) Create permissions for photo edits

- Create permissions that restrict access to creators of photo edits
- Use IsEditOwner permissions on PhotoEditViewSet

lewisemm committed:
(fix) Create tests for photo edit permissions

- Create tests for the IsEditOwner permissions

lewisemm committed:
(fix) Use less conspicous placeholder text

- Change font colour for the placeholder logo (when no photo is clicked)
to less conspicuous colour #eceff1. Also change font colour for placeholder
instructions
- Change font family for placeholder instructions to distinguish them
from the placeholder logo

lewisemm committed:
(fix) Fix malfunctioning permissions

- Fix list view of edits that displays all edits regardless of
whom the request came from

lewisemm committed:
Merge pull request #26 from andela-lkabui/fx-photo-edit-permissions

Fx photo edit permissions

lewisemm committed:
(chore) Use horizontal more items icon

- Use the horizontal icon on fixed action button to indicate that there
are more actions available when the button is hovered over.

lewisemm committed:
(chore) Update URLs

lewisemm committed:
Update readme

lewisemm committed:
Update gunicorn version

lewisemm committed:
Migrate to pipenv

lewisemm committed:
(chore) Fixes #28 Split django settings file

lewisemm committed:
(chore) Ignore direnv env var exports

lewisemm committed:
(chore) Convert tabs to spaces
